xla/service/gpu/gpu_windowed_einsum_handler.cc

```diff
@@ -48,16 +48,15 @@ namespace m = match;
 // and type conversions of FP8 operands into the bodies of their while loops,
 // i.e. rewrites
 //
-//   inputs --> dequant --> while loop {...}
+//   inputs --> dequant --> while loop {collective-permute/dot/etc}
 //
 // into
 //
-//   inputs --> while loop {dequant --> ...}.
-absl::StatusOr<bool> ShiftDequantizationF8(HloInstruction* instr) {
-  HloComputation* comp = instr->parent();
+//   inputs --> while loop {dequant --> collective-permute/dot/etc}.
+absl::Status ShiftDequantizationF8(HloComputation* comp) {
```

Comment:
Document `comp` is the body computation of the while-loop.

In fact, you get a `while_body` HloComputation* local variable below, which is exactly the same as `comp`. I would rename `comp` to `while_body` (which is obvious enough it doesn't have to be documented), and then remove the definition of `while_body` below.